projects/plugins/boost/app/assets/src/js/features/page-cache/health/health.tsx

```diff
@@ -1,48 +1,82 @@
-import { Notice } from '@automattic/jetpack-components';
-import getErrorData from './lib/get-error-data';
 import { usePageCacheSetup, type PageCacheError } from '$lib/stores/page-cache';
-import { useEffect, useState } from 'react';
+import { ReactNode, useEffect, useState } from 'react';
 import { useSingleModuleState } from '$features/module/lib/stores';
+import {
+	AdvancedCacheForSuperCacheNotice,
+	AdvancedCacheIncompatibleNotice,
+	FailedSettingsWriteNotice,
+	GenericErrorNotice,
+	NotUsingPermalinksNotice,
+	PageCacheRootDirNotWritableNotice,
+	UnableToWriteToAdvancedCacheNotice,
+	WPCacheDefinedNotTrueNotice,
+	WPConfigNotWritableNotice,
+	WPContentNotWritableNotice,
+} from './error-notices';
+import SwitchToBoost from '../switch-to-boost/switch-to-boost';
+
 type HealthProps = {
 	error?: PageCacheError;
-	setup?: ReturnType< typeof usePageCacheSetup >;
+	setup: ReturnType< typeof usePageCacheSetup >;
 	setError: ( error: PageCacheError ) => void;
 };
 
 const Health = ( { setup, error, setError }: HealthProps ) => {
-	const [ , setPageCache ] = useSingleModuleState( 'page_cache' );
+	const [ , setModuleState ] = useSingleModuleState( 'page_cache', state => {
+		// Run setup if the module is being enabled
+		if ( state ) {
+			setup.mutate();
+		}
+	} );
```

Comment:
This is a bit difficult to read. "If state exists, `setup.mutate()`?" That doesn't read like a prose - I need to look up the internals to understand what's going on. The comment tries to alleviate that, but I'm not sure what `module is being enabled` means

Reply:
I have updated the naming